src/device_trezor/device_trezor_base.cpp

```diff
@@ -475,20 +472,9 @@ namespace trezor {
       CHECK_AND_ASSERT_THROW_MES(msg, "Empty message");
       MDEBUG("on_passhprase_request");
 
-      // Backward compatibility, migration clause.
-      if (msg->has__on_device() && msg->_on_device()){
-        messages::common::PassphraseAck m;
-        resp = call_raw(&m);
-        return;
-      }
-
       m_seen_passphrase_entry_message = true;
       bool on_device = true;
-      if (msg->has__on_device() && !msg->_on_device()){
-        on_device = false;  // do not enter on device, old devices.
-      }
-
-      if (on_device && m_features && m_features->capabilities_size() > 0){
+      if (m_features && m_features->capabilities_size() > 0){
```

Comment:
If `capabilities_size()` is 0 (no capabilities at all), this code will leave `on_device` as true. But if there are some capabilities, but none of them are `Capability_PassphraseEntry`, it will set `on_device` to false. I don't know the specifications, but it doesn't seem logical to me.

Reply:
I think it's safe to remove `m_features->capabilities_size() > 0` check because if it's empty, the loop inside will not run any iterations, and `on_device` will be set to false.

Reply:
This has it's own history 😅 
https://docs.trezor.io/trezor-firmware/common/communication/passphrase-redesign-migration.html

tldr for quite a while there were no capabilities. Our firmware supports only Model TT, which has passphrase entry. Thus the default value=true. 

Then caps were added in new firmwares. When caps are there, we reset `on_device` to false and set back to true only if given cap is present. Thus, it made sense as old firmware -> no caps -> passphrase entry by default.

But now we dont support those old firmares anymore, so I will remove it.

Reply:
@SChernykh can I pls resolve this one?